### ISSUE TITLE
The kaxaml site is down

### ISSUE THREAD

**Issue by skimedic** [thinkpixellab/kaxaml#3]
Dear developers of Kaxaml.  The site [www.kaxaml.com](http://www.kaxaml.com) - it looks to be expired. I don't know if that was intentional or an over site, considering the waning popularity of WPF (although Xamarin is on the rise).
I have taken over rewriting Andrew Troelsen's book on Apress (Pro C#), splitting duties with Andrew on the last edition (7th), and now as a solo writer for the C# 7.1 version (8th edition). 
I use Kaxaml in the chapters on WPF since it's such an excellent tool for teaching and experimenting with XAML.  I would like to continue to use it in the book, so I wanted to reach out to you to see what my options are. Do you plan on reviving Kaxaml.com? If not, would you be opposed to me forking the repo and using that in the book?  Or I can just point out this repo in the book, but provide a compiled version for the readers to download.
It's the #1 book in Apress's .NET catalog, so I want to do right by the readers and right by you.
Thank you for your time and consideration.
Philip Japikse (@skimedic)

**Comment by bradzoob** [comment]
Yeah this seems like a pretty big deal, where can kaxaml be downloaded from now if we don't want to compile our own?

**Comment by kevlut** [comment]
Website is still down.  Please provide binaries in repo or bring site back! :)

**Comment by pankaj-nikam** [comment]
I believe that we should update this repo with the compiled binary. It is very difficult for now to download the setup and install Kaxaml. I am a big fan of Kaxaml!

**Comment by BHandle** [comment]
I have a built binary and a compiling version of the repository, but without the original download I can't tell what the license originally was.  Believe it or not, it doesn't say it in the program itself (other than it is free), and the word "license" does not appear in the original code.  Unfortunately, even codeplex doesn't have the download which might include the license, as the download was just a link to the kaxaml website.

I'd be happy to post my code once I figure out what that is.

*edit*  Ahh... but it did have the license.  MS-RL, so we're free to post derivatives.
http://kaxaml.codeplex.com/license

**Comment by RealityStop** [comment]
There is an active repo over at https://github.com/jogibear9988/kaxaml that includes an appveyor-built binary.  If that project is not actively maintained, I'll fork it.

**Comment by birbilis** [comment]
see https://ci.appveyor.com/project/jogibear9988/kaxaml/build/artifacts - that seems to have removed some Silverlight related stuff from menus etc., if you need older version I could search my backup for downloads, should have some

**Comment by OptikPerfectik** [comment]
Skimedic, thanks for looking into this.  I'm working through your book right now, and got stuck until I came to this page.  I just downloaded the old-fashioned rar file and opened the .sln in Visual Studio.  Regards.  It works, and wasn't too big of a hassle.

**Comment by skimedic** [comment]
@optik1234 Glad you were able to make it work.

**Comment by OptikPerfectik** [comment]
I should mention to users of the software that it throws and exception if you change the code on-the-fly in Kaxaml.  You have to make syntactically perfect code in Notepad++ or some other text editor and then cut and paste it in to avoid this error message.  If I were to make a code improvement suggestion, it would be to add a button, "OK to PARSE" so users won't have to keep going back and forth between a text editor and Kaxaml.  
Wow, actually this is kind of cool to be able to communicate with writer of a book I'm going through page by page.  5 stars sir!  It's very satisfying to see my skill-set getting better chapter by chapter.  Though I should be able to add the "OK to PARSE" button myself at this point.

**Comment by RealityStop** [comment]
You shouldn't need to, it should just grey out if it gets a xaml parse error.  Can you post your xaml code that triggers this?

**Comment by punker76** [comment]
Here is my version https://github.com/punker76/kaxaml. Adding dll references is coming soon. A first pre-releases is also available https://github.com/punker76/kaxaml/releases.

**Comment by RealityStop** [comment]
Aha, so you are continuing!  The WPF gods have smiled on us this day!

**Comment by OptikPerfectik** [comment]
To RealityStop:  It's every time that there is some syntactically incorrect Xaml.  I'd post an example, but I'm sure you can generate some incorrect code. :)

**Comment by punker76** [comment]
@Optik1234 This is the normal behavior / feature of this Tool. It try to parse the whole document in a timer, so if there is any incorrect input it produces a XamlParseException. You can only see this Exception in debug mode, if you use the release you don't see or feel it...

**Comment by punker76** [comment]
@Optik1234 @RealityStop @skimedic @birbilis @PixelHero @pankaj-nikam @kevlut Adding references finally available here https://github.com/punker76/kaxaml/releases

![kaxaml_matdesign1](https://user-images.githubusercontent.com/658431/33557342-45328516-d907-11e7-9253-1094fbd5210a.gif)

**Comment by ephermalite** [comment]
Good news, http://www.kaxaml.com/ is back up!

What? Don't give me that look. It is good news for Chinese citizens who enjoy playing games of chance on a site hosted in Philippines to circumvent rather strict gambling laws in China proper. And before you even mention it, shame on you for thinking a gambling site who's entire operation is based on deceit would tamper with the house edge.

**Comment by fforjan** [comment]
And down again !

**Comment by aesalazar** [comment]
If anyone needs binaries, I have some available on my fork.  This an upgrade to .net 9, bug fixes, and some added functionality include tag edit matching:

https://github.com/aesalazar/kaxaml/releases
